packages/app-builder/src/models/fuzzy-match.ts

```diff
@@ -0,0 +1,88 @@
+import { type TFunction } from 'i18next';
+import { assertNever } from 'typescript-utils';
+
+export const fuzzyMatchAlgorithms = [
+  'ratio',
+  'partial_ratio',
+  'token_sort_ratio',
+  'partial_token_sort_ratio',
+  'token_set_ratio',
+  'partial_token_set_ratio',
+] as const;
+export type FuzzyMatchAlgorithm = (typeof fuzzyMatchAlgorithms)[number];
+
+export function isFuzzyMatchAlgorithm(
+  value: string,
+): value is FuzzyMatchAlgorithm {
+  return (fuzzyMatchAlgorithms as ReadonlyArray<string>).includes(value);
+}
+
+export function getFuzzyMatchAlgorithmName(
+  t: TFunction<['common', 'scenarios'], undefined>,
```

Comment:
in the end t is not used, right ?

Reply:
don't we want to store those in the translation json files ?

Reply:
(this is where we could use "bag of words similarity" and "full string similarity", or something like this, for more human readable names)

Reply:
I added translation for both editable algorithm.
Just check I did not associated wrong translation to each algorithm, and if it's OK, this is mergeable.

Reply:
looks good !